addons/stock/models/stock_move.py

```diff
@@ -1046,14 +1005,14 @@ def _push_apply(self):
                 warehouse_id = False
 
             rule = ProcurementGroup._get_push_rule(move.product_id, move.location_dest_id, {
-                'route_ids': move.route_ids, 'product_packaging_id': move.product_packaging_id, 'warehouse_id': warehouse_id,
+                'route_ids': move.route_ids | move.move_line_ids.result_package_id.package_type_id.route_ids, 'warehouse_id': warehouse_id, 'uom_id': move.packaging_uom_id,
             })
 
             excluded_rule_ids = []
             while (rule and rule.push_domain and not move.filtered_domain(literal_eval(rule.push_domain))):
                 excluded_rule_ids.append(rule.id)
                 rule = ProcurementGroup._get_push_rule(move.product_id, move.location_dest_id, {
-                    'route_ids': move.route_ids, 'product_packaging_id': move.product_packaging_id, 'warehouse_id': warehouse_id,
+                    'route_ids': move.route_ids | move.move_line_ids.result_package_id.package_type_id.route_ids, 'warehouse_id': warehouse_id, 'uom_id': move.packaging_uom_id,
```

Comment:
Since the routes are defined on the packages (i.e. not packagings, thus not on the uom), does the part about `uom_id` still make sense? 

Also, you should leave the distinction between the routes explicitely set on the move (that were most probably set directly on a SO line). And having routes defined on a package type is not that simple at that point, because you can have a move that outputs to different packages (like one pallet and one box), each having different route settings.

In that case, you'd probably need a way to split your move based on the route to follow, and that's not something that his handled as far as I can see.